db/consistency_level.cc

```diff
@@ -257,6 +277,8 @@ filter_for_query(consistency_level cl,
         }
     }
 
+    validate_replica_dc_replication_factor(erm, live_endpoints);
```

Comment:
Why live endpoints only? It looks like the better place for the call would be `storage_proxy::get_endpoints_for_reading` before filtering out dead endpoints.

Reply:
ok.

Reply:
I wonder if this is the right approach. After all `storage_proxy` does a bunch of calculations and changes the set a bit, especially the filter_... function is pretty complex. What if there's a bug and we add an element to this set somewhere in these calcualtions -- in that case a later check would perhaps be more valuable. But I dunno.

Reply:
Well, the question is "what do we check her"? And the reason for the patch was a discrepancy between RF config and whatever erm returns, so here is the prefect place to check for it. If storage proxy has bugs it would not necessary be caught by this check. For instance the recent bug where because HWLB we had duplication in the set. The check would do nothing to catch it.